providers/core.tcpsocket.js

```diff
@@ -262,12 +286,25 @@ Socket_chrome.errorStringOfCode = function(code) {
       'UNKNOWN';
 };
 
+/**
+ * Get the Freedom error code associated with a Chrome error string
+ * @method freedomErrorCode
+ * @static
+ * @private
+ * @param {string} errorString The Chrome error string
+ * @return {string} The corresponding Freedom error string (often identical).
+ */
+Socket_chrome.freedomErrorCode = function(errorString) {
+  return Socket_chrome.FREEDOM_ERROR_MAP[errorString] || 'UNKNOWN';
+};
+
 /*
  * Alert a consumer that a socket is disconnected, with appropriate
  * error message.
  * @method dispatchDisconnect
  * @private
- * @param {Number} code the code returned by chrome when the socket closed.
+ * @param {Number=} code the code returned by chrome when the socket closed.
+ * @return {!Object} Error JSON object.
```

Comment:
could be documented better. Either
closure style record type: https://developers.google.com/closure/compiler/docs/js-for-compiler#records
or yui style: http://stackoverflow.com/questions/15396747/javascript-yuidoc-how-to-document-an-object-parameter-of-a-method/24795635#24795635